mettle/worker/worker.go

```diff
@@ -541,20 +556,26 @@ func (w *worker) extendAckDeadlineOnce(ctx context.Context, client *psraw.Subscr
 }
 
 // readRequest unmarshals the original execution request.
-func (w *worker) readRequest(msg []byte) (*pb.ExecuteRequest, *pb.Action, *pb.Command, *rpcstatus.Status) {
+func readRequest(msg []byte) (*pb.ExecuteRequest, error) {
 	req := &pb.ExecuteRequest{}
+	if err := proto.Unmarshal(msg, req); err != nil {
+		return nil, fmt.Errorf("Badly serialised request: %s", err)
+	}
+	return req, nil
+}
```

Comment:
Do we really need a separate function for this? Could we not just call `proto.Unmarshal` in the parent function?

Reply:
yea its not really adding any value